src/smartpeak/source/core/RawDataProcessor.cpp

```diff
@@ -2050,40 +2091,43 @@ namespace SmartPeak
+      std::vector<double> fragment_isotopomer_measured;
       
+      std::string fragment_isotopomer_theoretical_formula, fragment_isotopomer_measured_s;
+      double tmp, tmp2;
       for(auto& param : CalculateMDVAccuracies_params)
       {
         if (param.find("name")->second == "fragment_isotopomer_measured")
         {
-          if (!param.find("comment_")->second.empty())
+          if (!param.find("value")->second.empty())
           {
-            fragment_isotopomer_measured_s = param.find("comment_")->second;
-            std::stringstream fragment_isotopomer_measured_ss(fragment_isotopomer_measured_s);
-            std::string fragment_isotopomer_measured_buf;
-            while (fragment_isotopomer_measured_ss >> fragment_isotopomer_measured_buf)
-              fragment_isotopomer_measured.push_back(std::stod(fragment_isotopomer_measured_buf));
+            fragment_isotopomer_measured_s = param.find("value")->second;
+            std::regex regex_double("[+-]?\\d+(?:\\.\\d+)?");
+            std::sregex_iterator values_begin = std::sregex_iterator(fragment_isotopomer_measured_s.begin(), fragment_isotopomer_measured_s.end(), regex_double);
+            std::sregex_iterator values_end = std::sregex_iterator();
+            for (std::sregex_iterator it = values_begin; it != values_end; ++it)
+              fragment_isotopomer_measured.push_back(std::stod(it->str()));
           }
-        }
-        if (param.find("name")->second == "fragment_isotopomer_theoretical")
-        {
-          if (!param.find("comment_")->second.empty())
+          
+          if (!param.find("description")->second.empty())
           {
-            fragment_isotopomer_theoretical_s = param.find("comment_")->second;
-            std::stringstream fragment_isotopomer_theoretical_ss(fragment_isotopomer_theoretical_s);
-            std::string fragment_isotopomer_theoretical_buf;
-            while (fragment_isotopomer_theoretical_ss >> fragment_isotopomer_theoretical_buf)
-              fragment_isotopomer_theoretical.push_back(std::stod(fragment_isotopomer_theoretical_buf));
+            std::string fragment_isotopomer_id =param.find("description")->second;
+            
+            for (auto& peptide : rawDataHandler_IO.getTargetedExperiment().getPeptides())
+            {
+              if (peptide.metaValueExists("SumFormula") && peptide.id == fragment_isotopomer_id)
+              {
+                fragment_isotopomer_theoretical_formula = (std::string)peptide.getMetaValue("SumFormula");
+              }
+            }
           }
         }
       }
       
-      
       isotopelabelingmdvs.calculateMDVAccuracies(rawDataHandler_IO.getFeatureMap(), featureMap_with_accuracy_info,
-                                                 fragment_isotopomer_measured, fragment_isotopomer_theoretical);
+                                                 fragment_isotopomer_measured, fragment_isotopomer_theoretical_formula);
```

Comment:
Is this correct?  I am thinking that this should be a vector as there will be multiple sum formulas that match each of the different transitions in the TraML file (i.e., the getTargetedExperiment() object above).

Reply:
Yes, I have updated the paramter list to take in `feature_name` and a map of `proteinName`s to `SumFormula`s harvested from `OpenMSFile_traML_1.csv`.